mapper/amount.go

```diff
@@ -41,15 +47,21 @@ func Erc20Amount(
 	if isSender {
 		decimalValue = new(big.Int).Neg(decimalValue)
 	}
-	metadata := make(map[string]interface{})
-	metadata[ContractAddressMetadata] = contractAddress.String()
 
+	currency := Erc20Currency(contractSymbol, int32(contractDecimal), contractAddress.String())
 	return &types.Amount{
-		Value: decimalValue.String(),
-		Currency: &types.Currency{
-			Symbol:   contractSymbol,
-			Decimals: int32(contractDecimal),
-			Metadata: metadata,
-		},
+		Value:    decimalValue.String(),
+		Currency: currency,
+	}
+}
+
+func Erc20Currency(symbol string, decimals int32, contractAddress string) *types.Currency {
+	metadata := make(map[string]interface{})
+	metadata[ContractAddressMetadata] = contractAddress
```

Comment:
You can instantiate maps inline:
```golang
map[string]interface{}{
	ContractAddressMetadata: contractAddress,
}
```

Reply:
should also be moved into the return statement directly

Reply:
Not sure about Golang, but it's usually best to declare complex objects then return.  If an error happens in the return statement, the call stack can get messedup